Mailbox/Mailbox.cs

```diff
@@ -1,4 +1,41 @@
-﻿namespace Mailbox
+﻿using System;
+
+namespace Mailbox
 {
+    public class Mailbox
+    {
+        public Sizes MailSize { get; set; }
+        public (int x, int y) Location { get; set; }
+        public Person Owner { get; set; }
+
+        public Mailbox(Sizes mailSize, (int,int) location, Person owner)
+        {
+            if (mailSize == Sizes.Premium)
+            {
+                throw new ArgumentException("MailSize must have a premium size, not premium!");
+            }
+            MailSize = mailSize;
+            Location = location;
+            Owner = owner;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Mailbox) return this.Equals((Mailbox)obj);
```

Comment:
Pattern matching would be a little cleaner here.
```suggestion
            return obj is Mailbox mailbox and Equals(mailbox);
```